osu.Game/Online/API/APIAccess.cs

```diff
@@ -125,7 +125,6 @@ private void run()
                         userReq.Success += u =>
                         {
                             LocalUser.Value = u;
-                            Username = LocalUser.Value.Username;

```

Comment:
This doesn't seem intended?

Reply:
Easier to have User as dependency and get Username from him, or use LocalUser bindable they all return same without reassigment

Reply:
If the Username is not set it will continually set the API to offline? For example if requests failed because the API is unavailable and it tries to reconnect. Just wondering if that happens.

Reply:
No, it will go Offline and stay in it

Reply:
Oh I confused something then. Alright, thanks!

Reply:
Here https://github.com/ppy/osu/blob/master/osu.Game/Online/API/APIAccess.cs#L105

Reply:
Yes and `HasLogin` relies on the `Username` being present and I remember the API logging me out when it was having problems so I feared it wouldn't reconnect you. But then I also remembered that it was only a visual thing and the `Username` isn't actually cleared so it wouldn't matter. Hence I confused myself.